File: source/code/csharp/authorization/example_perm_complex.cs
var customData = await account.GetCustomDataAsync();

customData.Put("name", "createadmin");
customData.Put("description", "This permission allows the account to create an admin");
customData.Put("action", "read");
customData.Put("resource", "/admin/create");
customData.Put("effect", "allow");

await customData.SaveAsync();

File: source/code/nodejs/authorization/example_perm_complex.js
account.getCustomData(function (err, customData) {
  if (err) {
    return console.error(err);
  }

  customData.create_admin = true;
  customData.name = 'createadmin';
  customData.description = 'This permission allows the account to create an admin';
  customData.action = 'read';
  customData.resource = '/admin/create';
  customData.effect = 'allow';

  customData.save(function (err) {
    if (err) {
      return console.error(err);
    }

    console.log('Custom data updated.');
  })
});